api/controllers/requests-api-controller.ts

```diff
@@ -91,6 +91,11 @@ export const getAllRequests = async (req: Request, res: Response) => {
  */
 export const getRequestsByIDIR = async (req: Request, res: Response) => {
   const { minimal, idir } = req.query;
+
+  // Check if user's IDIR matches
+  const userInfo = getUserInfo(req.headers.authorization.split(' ')[1]); // Excludes the 'Bearer' part of token.
+  if (userInfo.idir_user_guid !== idir && !userInfo.client_roles?.includes('admin')) return res.status(403).send('Forbidden: User does not match requested IDIR.');
```

Comment:
Shouldn't this be an OR not and AND? Meaning they either match the idir or are an admin?

Reply:
If it were an OR, then a user whose IDIR matches but isn't an admin wouldn't be able to see their own record. 
An admin who is looking at another user's record would also be denied. 

I need this denial to only come if the user's IDIR doesn't match AND they aren't an admin. It's a weird double negative, but I think that's what I need. 

...

I ended up doing a truth table after thinking about your comment. It looks like I could also do:
```js
if (!(userInfo.idir_user_guid === idir || userInfo.client_roles?.includes('admin')))
```
But then I have the NOT the whole thing, which I find harder to read.

Reply:
There is definitely instances where using the (!( /* CONDITION */ )) makes sense and I think this is one of them.
You can also create

``` JavaScript
const idirMatchesUser = userInfo.idir_user_guid === idir;
const isAdmin = userInfo.client_roles?.includes('admin');

if (!(idirMatchesUser || isAdmin)) {}
```

Reply:
Actually that may not work. You might have to do if idir doesnt match and not an admin like you had it

Reply:
I ended up breaking down the conditions into separate variables for clarity, and comments were added, but I still think the`!X && !Y` logic is clearer than `!(X || Y)` in this case, so I've left that intact.

Reply:
@dbarkowsky wont that still not work for admins if the idir doesnt match?